src/read.c

```diff
@@ -485,7 +485,9 @@ void ReadCallVarAss (
     /* get the variable as a global variable                               */
     if ( type == ' ' ) {
         type = 'g';
-        var = GVarName( TLS(Value) );
+        /* we do not want to call GVarName on this value until after we
+         * have checked if this is the argument to a lambda function       */
+        strcpy(varname, TLS(Value));
```

Comment:
strcpy is fine here, as varname and TLS(Value) are arrays of the same length. However, it wouldn't hurt to use strlcpy for future protection